org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -232,18 +231,23 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
 		result.setCapabilities(createServerCapabilities(params));
 		access.addBuildListener(this);
 		return requestManager.runWrite(() -> {
-			if (workspaceManager.isSupportsWorkspaceFolders()) {
+			if (isWorkspaceFoldersSupported(params)) {
 				List<WorkspaceFolder> workspaceFolders = params.getWorkspaceFolders();
 				if (workspaceFolders == null)
 					workspaceFolders = Collections.emptyList();
 				workspaceManager.initialize(workspaceFolders, this::publishDiagnostics, CancelIndicator.NullImpl);
 			} else {
+				URI baseDir = getBaseDir(params);
 				workspaceManager.initialize(baseDir, this::publishDiagnostics, CancelIndicator.NullImpl);
 			}
 			return result;
 		}, (cancelIndicator, it) -> it).thenApply(it -> initializeResult = it);
 	}
 
+	private boolean isWorkspaceFoldersSupported(InitializeParams params) {
```

Comment:
should we make this protected?

Reply:
Could do that. Then we should use the class field initializeParams instead of requiring them as a method param.

Reply:
Yes, please